### DISCUSSION TITLE
How to solve thermal expansion problems with tens (or hundreds) of gas gap?

### DISCUSSION THREAD

**Discussion by Wang-Yihu** [idaholab/moose#29619] [Q&A Modules: Solid mechanics] [answered]
This geometry always appears in the solid nuclear reactor core: cylinder and hollow cylinders with the same axial line - rod, gas gap, clad, shown in this figure(top view):
![image](https://github.com/user-attachments/assets/d2c847dc-0ab3-4c4b-a0fe-e06d26fe9cf0)
I want to solve the thermal expansion for this problem. 
1) A natural thought is to only model rod and clad mesh. The mesh volumes remain void on the gas gap position. I use HeatConduction and HeatSource Kernels, [Modules/TensorMechanics/Master]. The temperature on the outer side of the rod or inner side of the clad is got from GapHeatTransfer. This method will be feasible for only one channel in this picture. But for tens or hundreds channels? It will cost lots of time which is unacceptable.
2) Another thought is to use different domains for Heat Conduction Module and Solid Mechanics Module: For Heat Conduction Module - rod, gas gap, clad; For Solid Mechanics Module:  rod, clad. But in this method, I encounter an error about negative Jacobian
![image](https://github.com/user-attachments/assets/d0bf7f5a-7a43-4a97-a90f-5ff726d2b3f0)
3) This method I cannot know the way to achieve it. Omit the gas gap for both rod and clad. In the Heat Conduction Module, try to make the temperature abrupt change to represent the thermal resistance caused by gas gap. In the Solid Mechanics Module, the displacement for the rod block and clad block is the same based on the assumption. However, I cannot find the way to achieve this....
So what's your suggestions? Do you know how to solve thermal expansion problems with tens (or hundreds) of gas gap, or find ways to achieve 3)?

**Comment by GiudGiud** [comment, marked as answer]
hello

this is likely covered in the BISON manual, which is available online
https://mooseframework.inl.gov/bison/
BISON is an application dedicated to these applications. I would recommend you ask for access here https://ncrcaims.inl.gov/Identity/Account/Login It might take a while to get access considering your current affiliation
